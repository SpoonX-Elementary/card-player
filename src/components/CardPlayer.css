.player {
  width: 854px;
  height: 480px;
  z-index: 1;
}
