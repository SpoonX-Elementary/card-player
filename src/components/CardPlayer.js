import React from 'react';
import PropTypes from 'prop-types';

export default class CardPlayer extends React.Component {
  render() {
    const { source, controls, style, poster } = this.props;

    return (
      <video controls={controls} style={style} poster={poster}>
        <source src={source} type="video/mp4" />
      </video>
    );
  }
}

CardPlayer.propTypes = {
  source  : PropTypes.any.isRequired,
  controls: PropTypes.bool,
  style   : PropTypes.object,
  poster  : PropTypes.string,
};

CardPlayer.defaultProps = {
  controls: true,
  style   : {},
  poster  : null,
};
