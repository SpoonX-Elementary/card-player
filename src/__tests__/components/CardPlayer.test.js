import React from 'react';
import renderer from 'react-test-renderer';
import CardPlayer from '../../components/CardPlayer';

test('Should render CardPlayer with default props', () => {
  const tree = renderer
    .create(<CardPlayer source="https://www.youtube.com/embed/gLRWloiIC58" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('Should render CardPlayer with custom props', () => {
  const tree = renderer
    .create(
      <CardPlayer
        source="https://www.youtube.com/embed/gLRWloiIC58"
        style={{ width: 720 }}
        poster="http://images5.fanpop.com/image/photos/30700000/Random-random-30798866-500-325.jpg"
        controls={false}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
