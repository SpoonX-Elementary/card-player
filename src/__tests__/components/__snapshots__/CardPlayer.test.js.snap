// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CardPlayer with custom props 1`] = `
<video
  controls={false}
  poster="http://images5.fanpop.com/image/photos/30700000/Random-random-30798866-500-325.jpg"
  style={
    Object {
      "width": 720,
    }
  }
>
  <source
    src="https://www.youtube.com/embed/gLRWloiIC58"
    type="video/mp4"
  />
</video>
`;

exports[`Should render CardPlayer with default props 1`] = `
<video
  controls={true}
  poster={null}
  style={Object {}}
>
  <source
    src="https://www.youtube.com/embed/gLRWloiIC58"
    type="video/mp4"
  />
</video>
`;
